jmespath-core/src/main/java/io/burt/jmespath/Adapter.java

```diff
@@ -32,6 +33,12 @@
   String toString(T value);
 
   /**
+   * Converts the argument to a Double, or null if the argument does not
+   * represent a number.
+   */
+  Double toDouble(T value);
```

Comment:
Boxed type because it needs to be able to return `null`, and `Double` and not `Number` because the primary use case is to pass the value into `Math.abs`, `Math.ceil`, etc. I guess that I could change it to `Number` and use `#doubleValue` before passing it into the math functions. I guess it's good to use the smallest interface possible, but it's also feels kind of like a mirage since the value will be converted to a double immediately either way. I might change it in the final review of the `Adapter` API.